src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -104,6 +112,41 @@ public async Task BackgroundUpdateAdvertisingManifestsWhenRequiredAsync()
             }
         }
 
+        private void WriteUpdatableWorkloadsFile()
+        {
+            var installedWorkloads = _workloadRecordRepo.GetInstalledWorkloads(_sdkFeatureBand);
+            var updatableWorkloads = GetUpdatableWorkloadsToAdvertise(installedWorkloads);
+            var filePath = GetAdvertisingWorkloadsFilePath();
+            var jsonContent = JsonSerializer.Serialize(updatableWorkloads.Select(workload => workload.ToString()).ToArray());
+            if (Directory.Exists(Path.GetDirectoryName(filePath)))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+            }
+            File.WriteAllText(filePath, jsonContent);
+        }
+
+        public static void AdvertiseWorkloadUpdates()
+        {
+            try
+            {
+                var backgroundUpdatesDisabled = bool.TryParse(Environment.GetEnvironmentVariable("DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_DISABLE"), out var disableEnvVar) && disableEnvVar;
+                var adUpdatesFile = Path.Combine(CliFolderPathCalculator.DotnetHomePath, ".dotnet", ".workloadAdvertisingUpdates");
+                if (!backgroundUpdatesDisabled && File.Exists(adUpdatesFile))
+                {
+                    var updatableWorkloads = JsonSerializer.Deserialize<string[]>(File.ReadAllText(adUpdatesFile));
+                    if (updatableWorkloads != null && updatableWorkloads.Any())
+                    {
+                        Console.WriteLine();
+                        Console.WriteLine(LocalizableStrings.WorkloadUpdatesAvailable);
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                // Never surface errors
```

Comment:
I think it would be a good idea to display the error if we are in debug (`-d` I think) or verbose logging mode.